File: Backend/TruckEase/TruckEase/Contracts/TransportRequest/Requests/CreateTransportRequest.cs
#nullable disable
using Newtonsoft.Json;
using TruckEase.Enums;

namespace TruckEase.Contracts.TransportRequest.Requests;

public class CreateTransportRequest
{
    [JsonProperty]
    public string Description { get; private set; }

    [JsonProperty]
    public DateTime StartTime { get; private set; }

    [JsonProperty]
    public DateTime ArrivalTime { get; private set; }

    [JsonProperty]
    public double Price { get; private set; }

    [JsonProperty]
    public string StartLocation { get; private set; }

    [JsonProperty]
    public string Destination { get; private set; }

    [JsonProperty]
    public bool IsExpress { get; private set; }

    [JsonProperty]
    public double LoadWeight { get; private set; }

    [JsonProperty]
    public bool IsDraft { get; private set; }

    [JsonProperty]
    public Transport TransportType { get; private set; }
}

File: Frontend/TruckEase/TruckEaseApp/src/app/models/requests/createTransportRequest.ts
import { Transport} from "../enums/transportType";

export class CreateTransportRequest {
    public Description!: string;
    public StartTime!: Date;
    public ArrivalTime!: Date;
    public StartLocation!: string;
    public Destination!: string;
    public IsExpress!: boolean;
    public Price!: DoubleRange;
    public LoadWeight!: DoubleRange;
    public IsDraft!: boolean;
    public TransportType!: Transport;
  }